internal/pkg/gateway/endpoint/endpoint.go

```diff
@@ -0,0 +1,186 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return &remoteEndorser{
+		client: peer.NewEndorserClient(conn),
+	}, nil
+}
+
+// NewAtomicBroadcastClient creates a connection to an orderer node.
+func (r *remote) NewAtomicBroadcastClient(url string, tlsRootCerts [][]byte) (Orderer, error) {
+	conn, err := r.conn.NewConnection(url, tlsRootCerts)
+	if err != nil {
+		return nil, err
+	}
+
+	return &orderer{
+		abc: ab.NewAtomicBroadcastClient(conn),
+	}, nil
+}
+
+type remoteEndorser struct {
+	client peer.EndorserClient
+}
+
+// ProcessProposal sends a signed proposal to the endorsing peer.
+func (r *remoteEndorser) ProcessProposal(ctx context.Context, in *peer.SignedProposal) (*peer.ProposalResponse, error) {
+	return r.client.ProcessProposal(ctx, in)
+}
+
+//go:generate counterfeiter -o mocks/atomicbroadcastclient.go --fake-name AtomicBroadcastClient . atomicBroadcastClient
+type atomicBroadcastClient interface {
+	ab.AtomicBroadcastClient
+}
+
+//go:generate counterfeiter -o mocks/broadcastclient.go --fake-name BroadcastClient . broadcastClient
+type broadcastClient interface {
+	ab.AtomicBroadcast_BroadcastClient
+}
+
+type orderer struct {
+	abc  atomicBroadcastClient
+	abbc broadcastClient
+}
+
+func (o *orderer) client() (broadcastClient, error) {
+	if o.abbc == nil {
+		abbc, err := o.abc.Broadcast(context.TODO()) // TODO build a context using timeouts specified in the gateway config (future user story)
+		if err != nil {
+			return nil, fmt.Errorf("failed to connect to orderer: %w", err)
+		}
+		o.abbc = abbc
+	}
+	return o.abbc, nil
+}
+
+func (o *orderer) Send(envelope *common.Envelope) error {
+	bc, err := o.client()
+	if err != nil {
+		return err
+	}
+	return bc.Send(envelope)
+}
+
+func (o *orderer) Recv() (*gatewayProto.Event, error) {
+	bc, err := o.client()
+	if err != nil {
+		return nil, err
+	}
+	response, err := bc.Recv()
+	if err != nil {
+		return nil, err
+	}
+	if err == io.EOF {
+		return nil, err
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to to get response from orderer: %w", err)
+	}
+	if response == nil {
+		return nil, fmt.Errorf("received nil response from orderer")
+	}
+
+	status := response.Info
+
+	return &gatewayProto.Event{
+		Value: []byte(status),
+	}, nil
+}
+
+//go:generate counterfeiter -o mocks/connectionProvider.go --fake-name ConnectionProvider . connectionProvider
+type connectionProvider interface {
+	NewConnection(url string, tlsRootCerts [][]byte) (*grpc.ClientConn, error)
+}
+
+type remoteConnection struct{}
+
+func (rc *remoteConnection) NewConnection(url string, tlsRootCerts [][]byte) (*grpc.ClientConn, error) {
+	secOpts := comm.SecureOptions{
+		UseTLS:            true,
+		ServerRootCAs:     tlsRootCerts,
+		RequireClientCert: false,
+	}
+	config := comm.ClientConfig{
+		SecOpts: secOpts,
+		Timeout: 5 * time.Second, // TODO from config
+	}
+	client, err := comm.NewGRPCClient(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return client.NewConnection(url)
```

Comment:
address != url and `NewConnection` takes an address.

Reply:
ok